woocommerce/woocommerce
# Allow experimental blocks to be enabled via the FeaturesController as part of removing WC_BLOCKS_PHASE

### Changes proposed in this Pull Request:

The overall goal of https://github.com/woocommerce/woocommerce/issues/43073 is to remove the concept of `WOOCOMMERCE_BLOCKS_PHASE` which has never really worked properly since the merge to the monorepo.

In the spirit of keeping PRs small and isolated this PR achieves https://github.com/woocommerce/woocommerce/issues/47692 which moves switching on/off of experimental blocks into the Advanced > Features menu. 

In a follow up PR I intend to remove `WC_BLOCKS_PHASE` altogether, by adding one additional setting to FeaturesController for enabling "Experimental block styling". With these 2 things in place we can remove all mentions of `buildPhase` or `WOOCOMMERCE_BLOCKS_PHASE` from the codebase.



### How to test the changes in this Pull Request:

1. You'll need WC beta tester installed, go to  Tools  > WCA Test Helper > Features and see that experimental blocks is enabled (in development)
2. Go to any block editor (page, post etc)
3. Search for an experimental block `Product Filter: Price`. You should see it's available in the inserter.
4. Repeat 1, but disable the setting (note that if you enabled or disabled the flag in a separate tab a page refresh is needed to ensure the setting updates) or do a production build
5. Repeat 3 but see that `Product Filter: Price` is no longer visible in the inserter



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add the ability to test experimental blocks via the Advanced > Features menu of WooCommerce settings.

#### Comment 

</details>
